test/params_api_test.c

```diff
@@ -103,12 +103,12 @@ static int test_param_type_extra(const OSSL_PARAM *param, unsigned char *cmp,
         if (signd) {
             if (!TEST_true(OSSL_PARAM_set_int32(param, 12345))
                 || !TEST_true(OSSL_PARAM_get_int64(param, &i64))
-                || !TEST_size_t_eq(i64, 12345))
+                || !TEST_size_t_eq((size_t)i64, 12345))
```

Comment:
For a 32 bit size_t this isn't testing anything.  Maybe it is time to add intmax tests and use them.
We've got definitions for intmax_t and uintmax_t already if the compiler doesn't supply them.  We don't have a printf modifier for them.

I won't block approval just for this.

Reply:
This just takes care of a compiler warning turned error...  rethinking is a matter for another PR, I'd say